### PR TITLE
Feat/new trip form and local storage

### PR DESCRIPTION
**Pull request by eduardoccmelo**

### CODE REVIEW

File: src/components/WorldMap.js [hunk 1]
```diff
@@ -7,12 +7,18 @@ import ReactMapGl, {
...
+  getMarkersFromLocalStorage,
+  removeMarkerFromLocalStorage,
+} from "../services/myMarkersStorage";
 
 export default function WorldMap() {
   const [countries, setCountries] = useState([]);
   const [visitedCountries, setVisitedCountries] = useState(0);
   const [filterInputValue, setFilterInputValue] = useState("");
   const [markers, setMarkers] = useState([]);
+  console.log(markers);
```
Comment (FriederikeM) [inline src/components/WorldMap.js:21]: Delete the console log before merging

File: src/components/WorldMap.js [hunk 2]
```diff
@@ -7,12 +7,18 @@ import ReactMapGl, {
...
 import "mapbox-gl/dist/mapbox-gl.css";
+import {
+  addMarkerToLocalStorage,
+  getMarkersFromLocalStorage,
+  removeMarkerFromLocalStorage,
+} from "../services/myMarkersStorage";
 
 export default function WorldMap() {
   const [countries, setCountries] = useState([]);
   const [visitedCountries, setVisitedCountries] = useState(0);
```
Comment (FriederikeM) [inline src/components/WorldMap.js:18]: maybe it's worth considering a name change here, as the value is a number, you could say numberOfVisitedCountries

when i read visitedCountries i expect an array with country names in it, even though i understand that's what's in the line above

**Review by FriederikeM — APPROVED** [review]
LGTM, just commented 2 small things

File: src/components/WorldMap.js (hunk 1 above)
Comment (eduardoccmelo) [reply src/components/WorldMap.js]: true

File: src/components/WorldMap.js (hunk 2 above)
Comment (eduardoccmelo) [reply src/components/WorldMap.js]: Yeah, it makes sense. Thanks Rikki

File: src/services/countriesCount.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  const myCountriesCount = getCountriesCountFromLocalStorage();
+  const newCount = Number(myCountriesCount) + 1;
+  localStorage.setItem("count", JSON.stringify(newCount));
+}
+
+export function removeCountryCountFromLocalStorage() {
+  const myCountriesCount = getCountriesCountFromLocalStorage();
+  const newCount = Number(myCountriesCount) - 1;
+  localStorage.setItem("count", JSON.stringify(newCount));
+}
```
Comment (mpagels) [inline src/services/countriesCount.js:16]: In our opinion the filename is a bit out of place. We would never expect localStorage functionality in here. You named your other files like `...MarkersStorage` what is way better.

File: src/services/myMarkersStorage.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  localStorage.setItem("markerData", JSON.stringify(myMarkers));
+}
+
+export function removeMarkerFromLocalStorage(tripName) {
+  const myMarkers = getMarkersFromLocalStorage();
+  const newMarkers = myMarkers.filter((marker) => {
+    return marker.name !== tripName;
+  });
+  localStorage.setItem("markerData", JSON.stringify(newMarkers));
+}
```
Comment (mpagels) [inline src/services/myMarkersStorage.js:18]: The file `myMarkersStorage.js` looks exactly the same as `myTripsStorage.js ` this screams for a refactoring. Basically you need to make the key, to which the data is saved to in localStorage, variable.

File: src/components/WorldMap.js [hunk 5]
```diff
@@ -48,33 +58,50 @@ export default function WorldMap() {
...
+      addMarkerToLocalStorage({ latlng, name });
+      addCountryCountToLocalStorage();
     } else {
-      setVisitedCountries(visitedCountries - 1);
+      setNumberOfVisitedCountries(numberOfVisitedCountries - 1);
       function filterFunction(country) {
-        return country !== latlng;
+        return country.name !== name;
       }
       const filteredMarkers = markers.filter(filterFunction);
```
Comment (mpagels) [inline src/components/WorldMap.js:78]: In our opinion this is hard to read. We thought the else body is done in line 77. Maybe you can use an arrow function in the filter method to get the confusion away.

File: src/components/WorldMap.js [hunk 6]
```diff
@@ -48,33 +58,50 @@ export default function WorldMap() {
...
+    textContent = "You didn't select any Country yet";
+  } else if (numberOfVisitedCountries === 1) {
+    textContent = `You have visited ${getCountriesCountFromLocalStorage()} Country in the World`;
+  } else {
+    textContent = `You have visited ${getCountriesCountFromLocalStorage()} Countries in the World`;
+  }
+
   function handleOnName(e) {
     setFilterInputValue(e.target.value);
   }
```
Comment (mpagels) [inline src/components/WorldMap.js:96]: That code snippet isn't used in the component.

**Review by mpagels — CHANGES_REQUESTED** [review]
Good work Eduardo. We've seen some overcomplicated code structure. That is not bad, because it gets the job done, but it has potential for a refactor. Maybe you should schedule some time for the next sprint to do that.

And don't forget to use your project board. 😉 

Yours sincerely truly respectfully best
M & Ms

File: src/components/WorldMap.js (hunk 6 above)
Comment (eduardoccmelo) [reply src/components/WorldMap.js:96]: I have to call this function in order to filter the name of the country:
``
 <input
        placeholder="Type the name of the country"
        className="filterInput"
        value={filterInputValue}
        id="filterInput"
        onChange={handleOnName}
      ></input>
``

**Pull request merged**